diff --git a/api.api.php b/api.api.php
index cdcf975..acc3b77 100644
--- a/api.api.php
+++ b/api.api.php
@@ -48,4 +48,16 @@ function hook_api_ignore_node_access($query) {
   return FALSE;
 }
 
-
+/**
+ * Alter the list of API parse functions.
+ *
+ * @param array $parse_functions
+ *   An associative array whose keys are the file extensions to parse, and
+ *   whose values are the callback functions used to parse them. See
+ *   api_parse_functions() for details on the callback function.
+ *
+ * @see api_parse_functions()
+ */
+function hook_api_parse_functions_alter(array &$parse_functions) {
+  $parse_functions['php'] = 'my_module_api_parse_php_file';
+}
diff --git a/parser.inc b/parser.inc
index 9420ccf..602652c 100644
--- a/parser.inc
+++ b/parser.inc
@@ -3099,11 +3099,24 @@ function api_should_queue($branch) {
  *
  * @return
  *   An associative array whose keys are file extensions, and whose values are
- *   the functions used to parse files with that extension.
+ *   the functions used to parse files with that extension. The function should
+ *   parse the contents of the file, and save one or more documentation objects
+ *   in the database. The function argument is a $docblock array, containing a
+ *   sample documentation listing for the file as a whole. In this array, the
+ *   following keys are filled in:
+ *   - object_name, file_name: Both set to the file name.
+ *   - title, basename: Both set to the file basename.
+ *   - modified: Set to the time the file was last modified.
+ *   - branch: The branch object.
+ *   - object_type: Set to 'file'.
+ *   - source: Set to the file contents.
+ *   - is_drupal: TRUE if this is a Drupal file, as defined by the regular
+ *     expression for deciding this that is part of the branch setup.
+ *
+ * @see api_save_documentation()
  */
 function api_parse_functions() {
-  // @todo This could use a hook to let other modules extend this array?
-  return array(
+  $functions = array(
     'php' => 'api_parse_php_file',
     'module' => 'api_parse_php_file',
     'inc' => 'api_parse_php_file',
@@ -3124,6 +3137,9 @@ function api_parse_functions() {
     'htm' => 'api_parse_html_file',
     'html' => 'api_parse_html_file',
   );
+
+  drupal_alter('api_parse_functions', $functions);
+  return $functions;
 }
 
 /**
