From e1c5c0556f13bd067f2a8bb499e034e4d50e9c43 Mon Sep 17 00:00:00 2001
From: Neil Drumm
Date: Thu, 9 Jun 2016 10:13:54 -0400
Subject: Issue #2745593 by karoop: PHP version check fails with PHP 7

---
 api.info   | 2 +-
 api.module | 9 ---------
 2 files changed, 1 insertion(+), 10 deletions(-)

diff --git a/api.info b/api.info
index 6abd4db..98494fc 100644
--- a/api.info
+++ b/api.info
@@ -3,7 +3,7 @@ description = Generates and displays API documentation pages.
 package = Development
 core = 7.x
 configure = admin/config/development/api
-php = 5.3
+php = 5.4
 dependencies[] = node
 dependencies[] = ctools
 dependencies[] = gplib (>=2.1,<3.0)
diff --git a/api.module b/api.module
index 99f6f1e..6b61f50 100644
--- a/api.module
+++ b/api.module
@@ -3312,15 +3312,6 @@ function api_libraries_loaded($watchdog = TRUE) {
     return FALSE;
   }
 
-  $version = explode('.', PHP_VERSION);
-  if ($version[0] < 5 || $version[1] < 4) {
-    if ($watchdog) {
-      watchdog('api', 'This version of the API module requires PHP 5.4 or later for parsing', WATCHDOG_ERROR);
-    }
-    $loaded = FALSE;
-    return FALSE;
-  }
-
   $loaded = TRUE;
   return TRUE;
 }
-- 
cgit v0.12

