diff --git a/composer_manager.admin.inc b/composer_manager.admin.inc
index 5c9c29e..bb43da5 100644
--- a/composer_manager.admin.inc
+++ b/composer_manager.admin.inc
@@ -32,10 +32,10 @@ function composer_manager_settings_form($form, &$form_state) {
   );
 
   $form['composer_manager_autobuild_file'] = array(
-    '#title' => 'Automatically build the composer.json file when enabling or disabling modules in the Drupal UI',
+    '#title' => 'Automatically build the composer.json file when enabling or disabling modules',
     '#type' => 'checkbox',
     '#default_value' => variable_get('composer_manager_autobuild_file', 1),
-    '#description' => t('Automatically build the consolidated composer.json for all contributed modules file in the vendor directory above when modules are enabled or disabled in the Drupal UI. Disable this setting if you want to maintain the composer.json file manually.'),
+    '#description' => t('Automatically build the consolidated composer.json for all contributed modules file in the vendor directory above when modules are enabled or disabled. Disable this setting if you want to maintain the composer.json file manually.'),
   );
 
   $form['composer_manager_autobuild_packages'] = array(
diff --git a/composer_manager.module b/composer_manager.module
index 3995eed..a842a22 100644
--- a/composer_manager.module
+++ b/composer_manager.module
@@ -88,17 +88,16 @@ function composer_manager_modules_disabled($modules) {
  * has a composer.json file.
  *
  * This is a primitive check to ensure that the composer.json file is built only
- * when it has changes. This check always passes when run via the command line,
- * as it is assumed that Drush is being used to enable or disable the the
- * modules. A static boolean is also set flagging whether one or more modules
- * being acted on contains a composer.json file, which is used in Drush hooks.
+ * when it has changes. A static boolean is also set flagging whether one or
+ * more modules being acted on contains a composer.json file, which is used in
+ * Drush hooks.
  *
  * @param array $modules
  *   The enabled / disabled modules being scanned for a composer.json file.
  */
 function composer_manager_write_if_changed(array $modules) {
   $changed = &drupal_static(__FUNCTION__, FALSE);
-  if (variable_get('composer_manager_autobuild_file', 1) || drupal_is_cli()) {
+  if (variable_get('composer_manager_autobuild_file', 1)) {
     if (composer_manager_packages_have_changed($modules)) {
       $changed = TRUE;
       composer_manager_write_file();
